### DISCUSSION TITLE
Brew install show already downloaded and stuck forever...

### DISCUSSION THREAD

**Discussion by Nauxscript** [Homebrew/discussions#3457] [Casks]
<img width="1022" alt="image" src="https://user-images.githubusercontent.com/13010589/180342200-9a2ae520-6171-4272-a152-a9e2fab928be.png">

just like above screen shot;

this has happened several times recent downloading app; I wonder how to fixed it or just let the installation executing normally

**Comment by alebcay** [comment]
At a glance, I'm not sure either why it may be getting stuck here, but can you please try again with `--verbose` and `--debug` flags? This should provide more information to make troubleshooting easier.

**Reply by Nauxscript** [reply]
ok i will try it

**Reply by carlocab** [reply]
By the way, please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse (e.g. when looking up error messages). Instead, copy the text here and wrap them in three back-ticks (```) on the line before and on the line after to format them properly.

**Reply by Nauxscript** [reply]
i see that, thanks for you useful suggestion

**Reply by Nauxscript** [reply]
today this situation come again when i install obsidian, its just show Already downloaded and stuck forever (maybe 4 ~ 5 hours), and i stop it in the end, command output：
```
brew install obsidian --cask
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
berkeley-db@5                       gum                                 pkcs11-tools
c2rust                              hyx                                 prql-compiler
cargo-depgraph                      jupp                                prr
chain-bench                         ksh93                               sgr
cpp-httplib                         libeatmydata                        smap
create-api                          mesheryctl                          snapcast
dura                                onlykey-agent                       tere
felinks                             open62541                           tlsx
gaze                                page                                tuntox
git-delete-merged-branches          pint                                x86_64-linux-gnu-binutils

You have 7 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

==> Downloading https://github.com/obsidianmd/obsidian-releases/releases/download/v0.15.6/Obsidian-0.15.6-u
Already downloaded: /Users/naux/Library/Caches/Homebrew/downloads/648052cece8a52e821dc55e63edb991980f76275c572d018be5654dc4a21e446--Obsidian-0.15.6-universal.dmg
``` 

and i try again with `--verbose` and `--debug` flags:
```
brew install obsidian --cask --verbose
==> Downloading https://github.com/obsidianmd/obsidian-releases/releases/download/v0.15.6/Obsidian-0.15.6-universal.dmg
Already downloaded: /Users/naux/Library/Caches/Homebrew/downloads/648052cece8a52e821dc55e63edb991980f76275c572d018be5654dc4a21e446--Obsidian-0.15.6-universal.dmg
```

```
brew install obsidian --cask --debug
==> Cask::Installer#install
==> Printing caveats
==> Cask::Installer#fetch
==> Downloading https://github.com/obsidianmd/obsidian-releases/releases/download/v0.15.6/Obsidian-0.15.6-u
/opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.5.7\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 12.2.1\)\ curl/7.77.0 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://github.com/obsidianmd/obsidian-releases/releases/download/v0.15.6/Obsidian-0.15.6-universal.dmg
Already downloaded: /Users/naux/Library/Caches/Homebrew/downloads/648052cece8a52e821dc55e63edb991980f76275c572d018be5654dc4a21e446--Obsidian-0.15.6-universal.dmg
==> Checking quarantine support
/usr/bin/xattr -h
/usr/bin/swift -target arm64-apple-macosx12 /opt/homebrew/Library/Homebrew/cask/utils/quarantine.swift
```

nothing happen but stuck.

**Comment by marzsv** [comment]
I was able to successfully install all the necessary packages by using a VPN to connect to a US server. With the VPN in place, I experienced no further issues with the download process.

**Comment by ssghost** [comment]
I have the same problem here on Mac OSX 12.7.6 Homebrew 4.4.19-21-g9b6de82

**Reply by ssghost** [reply]
btw, I am in China, and the problem is always the same whether I turned on or off the VPN.

**Reply by gromgit** [reply]
I believe there are mainland mirrors hosted by various universities, like this one: https://mirrors.tuna.tsinghua.edu.cn/help/homebrew/
